src/track/track.h

```diff
@@ -212,6 +211,26 @@ class Track : public QObject {
     void setTrackNumber(const QString&);
     void setTrackTotal(const QString&);
 
+    /// Return the genre as text
+    QString getGenreText() const;
+    /// !!!DO NOT USE!!!
+    /// Set the genre text WITHOUT updating the corresponding custom tags.
+    ///
+    /// Only allowed to be used by TrackDAO!!! Unfortunately, the
+    /// design of TrackDAO does not allow to hide this method by
+    /// making it private.
+    void setGenreTextInternal(
+            const QString& genreText);
+    bool updateGenreText(
```

Comment:
Is this method is the replacement for `setGenreText`? Can you elaborate why it's named differently and not exposed as a setter in `QmlPlayerProxy`?

Reply:
...because we will need `TrackCollectionManager` for the custom tag mapping config. This context is not available in individual track objects.